src/parser/base.rs

```diff
@@ -212,11 +215,24 @@ impl<'a> Parser<'a> {
         }
     }
 
-    fn read_end(&mut self) {
+    fn read_end(&mut self) -> Option<()> {
         self.stream.advance();
+
+        let closing_tag_name = self.read_ident()?;
+
         self.read_to(b'>');
         self.stream.expect_and_skip_cond(b'>');
 
+        let closing_tag_matches_parent = self.stack.last()
+            .and_then(|last_handle| last_handle.get(self))
+            .and_then(|last_item| last_item.as_tag())
+            .and_then(|last_tag| Some(last_tag.name() == closing_tag_name))
+            .unwrap_or(false);
```

Comment:
Nit: those two lines could be written as 
```suggestion
            .map_or(false, |last_tag| last_tag.name() == closing_tag_name);
```